### PR TITLE
Skip intro screen

### PR DESCRIPTION
**Pull request by max2me**
Intro screen will be shown only once unless form has customized logo. In that case it will always be shown.

### CODE REVIEW

**Comment by max2me** [comment]
Spacing issues are getting a bit out of hand. I'll submit another PR that fixes spacing issues on the entire repo.

File: collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java [hunk 1]
```diff
@@ -1255,7 +1055,234 @@ public void onClick(View v) {
...
+        }
+
+        return endView;
+    }
+
+    private View createFormBeginningView(FormController formController, boolean advancingPage) {
+        View startView = View.inflate(this, R.layout.form_entry_start, null);
+        setTitle(getString(R.string.app_name) + " > " + formController.getFormTitle());
+
+		if (shouldShowIntroScreenBecauseOfLogo(formController)) { // If image is specified, we will always show intro screen
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java]: Let's follow the Android project style guide for now and [keep lines short](http://source.android.com/source/code-style.html#limit-line-length). For conditionals, that means commenting on the next line unless you just have something very short to say:

```
if () {
    // long comment here
    ...
} else {
    // long comment here
    ...
}
```

File: collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java [hunk 2]
```diff
@@ -1255,7 +1055,234 @@ public void onClick(View v) {
...
+		return getFormLogo(formController) != null;
+	}
+
+	private Drawable getFormLogo(FormController formController) {
+        Drawable image = null;
+        File mediaFolder = formController.getMediaFolder();
+        String mediaDir = mediaFolder.getAbsolutePath();
+        BitmapDrawable bitImage = null;
+        // attempt to load the form-specific logo...
+        // this is arbitrarily silly
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java]: This kind of comment would be good to address and/or remove (the "arbitrarily silly" part).

**Review by lognaturel — CHANGES_REQUESTED** [review]
This is great! Looks like it wasn't such a huge change after all.

Little thing to add - there needs to be a user-editable general setting called "Show form start screen" that goes right after the app splash screen setting. This should make it so that the splash screen is always displayed (behavior of the current released version). @yanokwa and I briefly discussed this and thought it should default to on to maintain existing behavior which works for most people. Open to discussion around that, though.

There also needs to be an admin setting that shows or hides the setting described above. You'll see that the other settings follow this pattern.

**Comment by lognaturel** [comment]
Here's a quick procedural thing -- let's try to be consistent about using [keywords for closing issues](https://help.github.com/articles/closing-issues-via-commit-messages/) in commit messages and/or PR descriptions so there are beautiful referrals between code and issues and the issues get closed automatically on merge. Like this: closes #182 😊

**Comment by yanokwa** [comment]
@max2me I've put the spacing issue at https://github.com/opendatakit/collect/issues/231

**Comment by max2me** [comment]
@lognaturel I've formatted comment and rewritten `getFormLogo()` method. 

**Just to clarify:**
1. Should 'closes' keyword be used in individual commits or as a description of pull request?
2. I suspect there is value in squashing commits when merging into a single one. If you agree, then we also need to reference issue in that new commit message. What do you think?

**Re: functionality:**
Let me ask a few questions to better understand intent here. What's the goal of that screen when there is no logo specified? To educate users on how to navigate. My intuition is that after people have went through the survey once (since they have to swipe to get between questions) they will remember how to do it and dont need additional prompt which is just getting in their way.

We can definitely add settings etc but as designers / project maintainers it is our job to strike the balance between flexibility & usability of the product as a whole. Settings are great when there is not a single solution/behavior that works for everyone (or even for the vast majority of users) but if we start adding too many settings / customizations, users might have a hard time navigating the app or even discovering settings that truly matter to them.

So I guess my question is - are there scenarios where someone needs prompting every time they start filling out a form? I'm happy to add both settings in either way but just wondering...

**Comment by lognaturel** [comment]
I personally am not picky about where the issue closing is triggered from. I would say do what makes sense to you and as there are more contributors we can decide whether or not it's something to define more precisely. Commits have not so far been squashed on merge and I think that's another thing that can be revisited as there are more contributors and committers!

My understanding is that individual data collectors don't typically do much with settings. What generally happens is that an organization will provide certain defaults on all their phones and use the admin settings to decide what, if anything, they let their data collectors change. Data collectors are trained in a rote way and may not have any other experience using smartphones so in general it's better not to change existing functionality much even if it seems trivial. @yanokwa?

**Comment by lognaturel** [comment]
On further reflection, showing it only once doesn't make sense. The devices aren't single-user so often it would just be the IT staff that would see it rather than a data collector. So the question is whether to have a setting or to do away with it all together. It's a good question for the community so @yanokwa will ask the mailing list along with the analytics question.

**Comment by yanokwa** [comment]
We've had a bit of a discussion at https://groups.google.com/d/msg/opendatakit/MlNVA5UCaUk/3YxMXnI4BwAJ and I think we should remove the screen entirely. No settings.

**Comment by max2me** [comment]
I'm on it!

**Comment by lognaturel** [comment]
I think this breaks forms that start with a group. Try out "Biggest N of Set" from the test server as an example. The form should start out by asking whether a group should be added.

Let's punt on this for today's release so we have time to fix this and really make sure it doesn't break anything else.

**Comment by max2me** [comment]
Closing this. Will port changes to a new branch for easier code review and will file new PR then.

**Comment by lognaturel** [comment]
New and exciting pull request at #274

**Pull request closed without merging**